test/hotspot/jtreg/runtime/cds/appcds/jcmd/JCmdTest.java

```diff
@@ -0,0 +1,364 @@
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+/*
+ * @test
+ * @bug 8259070
+ * @summary Test jcmd to dump static and dynamic shared archive.
+ * @requires vm.cds
+ * @library /test/lib /test/hotspot/jtreg/runtime/cds/appcds
+ * @modules jdk.jcmd/sun.tools.common:+open
+ * @compile ../test-classes/Hello.java
+ * @build sun.hotspot.WhiteBox
+ * @build JCmdTestLingeredApp JCmdTest
+ * @run driver jdk.test.lib.helpers.ClassFileInstaller sun.hotspot.WhiteBox
+ * @run main/othervm/timeout=480 -Xbootclasspath/a:. -XX:+UnlockDiagnosticVMOptions -XX:+WhiteBoxAPI JCmdTest
+ */
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.nio.file.Files;
+import java.nio.file.FileSystems;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+
+import jdk.test.lib.apps.LingeredApp;
+import jdk.test.lib.cds.CDSTestUtils;
+import jdk.test.lib.dcmd.PidJcmdExecutor;
+import jdk.test.lib.process.OutputAnalyzer;
+import jdk.test.lib.Utils;
+import jtreg.SkippedException;
+import sun.hotspot.WhiteBox;
+
+import java.io.InputStreamReader;
+import java.io.BufferedReader;
+
+public class JCmdTest {
+    static final String TEST_CLASSES[]      = {"JCmdTestLingeredApp",
+                                               "jdk/test/lib/apps/LingeredApp",
+                                               "jdk/test/lib/apps/LingeredApp$1"};
+    static final String BOOT_CLASSES[]      = {"Hello"};
+    static final String SUBCMD_STATIC_DUMP  = "static_dump";
+    static final String SUBCMD_DYNAMIC_DUMP = "dynamic_dump";
+
+    static final String STATIC_DUMP_FILE    = "mystatic";
+    static final String DYNAMIC_DUMP_FILE   = "mydynamic";
+
+
+    static final String[] STATIC_MESSAGES   = {"JCmdTestLingeredApp source: shared objects file",
+                                               "LingeredApp source: shared objects file",
+                                               "Hello source: shared objects file"};
+    static final String[] DYNAMIC_MESSAGES  = {"JCmdTestLingeredApp source: shared objects file (top)",
+                                               "LingeredApp source: shared objects file (top)",
+                                               "Hello source: shared objects file (top)"};
+
+    static String testJar = null;
+    static String bootJar = null;
+    static String allJars = null;
+
+    private static void buildJar() throws Exception {
+        testJar = JarBuilder.build("test", TEST_CLASSES);
+        bootJar = JarBuilder.build("boot", BOOT_CLASSES);
+        Path testJarPath = FileSystems.getDefault().getPath(testJar);
+        Path bootJarPath = FileSystems.getDefault().getPath(bootJar);
```

Comment:
I think the above could be replaced with:
```
    String testJarPath = TestCommon.getTestJar(“test.jar”);
    String bootJarPath = TestCommon.getTestJar(“boot.jar”);
```
Most of the CDS tests use the `getTestJar` method.
You can leave it as is if you like.